### PR TITLE
My Site Dashboard (Phase 2): Add ability to remove Quick Start from dashboard

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Part of #17877

## Description
- Refactored code related to removing next steps from `BlogDetailsViewController` to a category on `UIViewController` so it can be used in the dashboard as well
- Added an ellipsis button to `BlogDashboardCardFrameView`
- Used newly added button in quick start card to trigger the "remove next steps" action sheet

 

https://user-images.githubusercontent.com/25306722/158922347-dd35d1ba-c8be-4e13-b483-7c6b9eef91db.mov

### Remove Next Steps from the dashboard
1. Make sure the MSD feature flag is enabled
2. Go into app settings and set the initial screen to "Home"
3. Enable quick start for your site
4. Observe the quick start card in the dashboard, an ellipsis button should be displayed
5. Make sure the a11y is correctly set for the button
6. Tap the ellipsis button
7. Make sure an action sheet is displayed
8. Go through the alert to remove "Next steps"
9. Make sure the quick start card is removed.

### Regression on removing Next Steps from My Site
1. Make sure the MSD feature flag is enabled
2. Go into app settings and set the initial screen to "Site Menu"
3. Enable quick start for your site
4. Observe the quick start section in My Site
5. Tap the ellipsis button
6. Make sure an action sheet is displayed
7. Go through the alert to remove "Next steps"
8. Make sure the quick start section is removed.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by hassaanelgarem** [comment]
@osullivanchris I couldn't find anything in the designs related to this task (Removing Next Steps from dashboard), so I improvised a bit. Please take a look and let me know what you think. I'm open to any suggestions 👍

**Comment by osullivanchris** [comment]
> @osullivanchris I couldn't find anything in the designs related to this task (Removing Next Steps from dashboard), so I improvised a bit. Please take a look and let me know what you think. I'm open to any suggestions 👍

Hey @hassaanelgarem I'm pretty happy with this one. From what I can tell you've re-used code from how a user removes the quick steps from the menu already, is that right?

The only consideration is that other cards could have an elipsis and secondary actions in future. Maybe its something we'll in some way standardise down the line.

@iamthomasbishop who is doing design work for a similar action on another card. FYI here is an example of a card with secondary action. I know you were looking at Android. But here's how one is already working on iOS. I haven't done any custom work on it. Its just derived from what we already had for the existing Quick Start.

File: WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift [hunk 1]
```diff
@@ -44,6 +44,7 @@ open class QuickStartTourGuide: NSObject {
 
     @objc func remove(from blog: Blog) {
         blog.removeAllTours()
+        NotificationCenter.default.post(name: .QuickStartTourElementChangedNotification, object: self)
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift:48]: What do you think about changing `BlogDashboardViewController.showQuickStart` to `toggleQuickStart`? Or `showOrHideQuickStart`?

**Review by leandroalonso — APPROVED** [review]
Tested multiple cases:

1. With MSD flag disabled and the current flow
2. With the flag enabled and Site Menu as default
3. With the flag enabled and Home as default
4. Starting Quick Start from settings and epilogue

One minor thing: if you remove Quick Start between steps, it gets into a weird state. Eg.:

1. Start Quick Start
2. Start "Check your site title"
3. Dismiss Quick Start
4. Tap the title
5. Quick Start is back!

I tested the current version and I think this behavior has been there for quite some time (and I think fixing it is not so simple). Given that let's go ahead with this PR and we can track this on an issue.

**Comment by hassaanelgarem** [comment]
> From what I can tell you've re-used code from how a user removes the quick steps from the menu already, is that right?

@osullivanchris Yes, that is correct.

> The only consideration is that other cards could have an elipsis and secondary actions in future. Maybe its something we'll in some way standardise down the line.

The elipsis is already implemented in a generic way, so it can be easily adopted by other cards. As for the action sheet, this is not standardized, but we can tackle that in the future 👍

File: WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift (hunk 1 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift:48]: Agree! Fixed in 10026ed3a317a3c6124425e934f73b8e7253019c

**Comment by hassaanelgarem** [comment]
> One minor thing: if you remove Quick Start between steps, it gets into a weird state. 

@leandroalonso Nice catch! The fix was actually simpler than I anticipated, so I decided to include it in this PR.
Fixed in 62f8915bd08dc899bce265492b8616c160fbf3e0

**Comment by leandroalonso** [comment]
@hassaanelgarem amazing! Tested and it works! :shipit:

**Pull request merged**